public/app/plugins/datasource/prometheus/components/PromQueryField.test.tsx

```diff
@@ -51,61 +51,89 @@ describe('PromQueryField', () => {
   });
 
   it('refreshes metrics when the data source changes', async () => {
+    const defaultProps = {
+      query: { expr: '', refId: '' },
+      onRunQuery: () => {},
+      onChange: () => {},
+      history: [],
+    };
     const metrics = ['foo', 'bar'];
-    const languageProvider = ({
-      histogramMetrics: [] as any,
-      metrics,
-      metricsMetadata: {},
-      lookupsDisabled: false,
-      lookupMetricsThreshold: DEFAULT_LOOKUP_METRICS_THRESHOLD,
-      start: () => {
-        return Promise.resolve([]);
-      },
-    } as unknown) as PromQlLanguageProvider;
-
     const queryField = render(
       <PromQueryField
         // @ts-ignore
         datasource={{
-          languageProvider,
+          languageProvider: makeLanguageProvider({ metrics: [metrics] }),
         }}
-        query={{ expr: '', refId: '' }}
-        onRunQuery={() => {}}
-        onChange={() => {}}
-        history={[]}
+        {...defaultProps}
       />
     );
 
-    let cascader = await queryField.findByRole('button');
-    fireEvent.keyDown(cascader, { keyCode: 40 });
-    let listNodes = screen.getAllByRole('menuitem');
-    for (const node of listNodes) {
-      expect(metrics).toContain(node.innerHTML);
-    }
+    checkMetricsInCascader(await screen.findByRole('button'), metrics);
 
     const changedMetrics = ['baz', 'moo'];
     queryField.rerender(
       <PromQueryField
+        // @ts-ignore
         datasource={{
-          //@ts-ignore
-          languageProvider: {
-            ...languageProvider,
-            metrics: changedMetrics,
-          },
+          languageProvider: makeLanguageProvider({ metrics: [changedMetrics] }),
         }}
-        query={{ expr: '', refId: '' }}
-        onRunQuery={() => {}}
-        onChange={() => {}}
-        history={[]}
+        {...defaultProps}
       />
     );
 
-    cascader = await queryField.findByRole('button');
-    fireEvent.keyDown(cascader, { keyCode: 40 });
-    listNodes = screen.getAllByRole('menuitem');
-    for (const node of listNodes) {
-      expect(changedMetrics).toContain(node.innerHTML);
-    }
+    // If we check the cascader right away it should be in loading state
+    let cascader = screen.getByRole('button');
+    expect(cascader.textContent).toContain('Loading');
+    checkMetricsInCascader(await screen.findByRole('button'), changedMetrics);
+  });
+
+  it('refreshes metrics when time range changes but dont show loading state', async () => {
```

Comment:
Testing slate itself is tricky so this is just indirect test and just checks if we get to loading state after timeRange change